inference/marian-fork/src/common/binary.cpp

```diff
@@ -65,29 +81,44 @@ void loadItems(const void* current, std::vector<io::Item>& items, bool mapped) {
                               // Also when using the full mmap codepath, we need to uncomment expression_graph.h:582
-    uint64_t len = headers[i].dataLength;
-    items[i].bytes.resize(len);
-    const char* ptr = get<char>(current, len);
-    if (matchType<intgemm8>(items[i].type)) {
-      if (items[i].name.find("Wemb") != std::string::npos) { // Since Wemb need to be dequantised, we have a special case for them
+
+    auto resize = [&](uint64_t len) {
+      items[i].bytes->resize(len);
+      totalBytesLoaded += len;
+    };
+    const char* ptr = get<char>(current, headers[i].dataLength);
+
+    if(matchType<intgemm8>(items[i].type)) {
+      if(items[i].name.find("Wemb") != std::string::npos) {  // Since Wemb need to be dequantised,
+                                                             // we have a special case for them
         items[i].type = Type::float32;
-        items[i].bytes.resize(items[i].shape.elements()*sizeof(float)); // We should have an extra float at the back but that requires a different format, due to allocator work
+        resize(items[i].shape.elements()
+               * sizeof(float));  // We should have an extra float at the back but that requires a
+                                  // different format, due to allocator work
         cpu::integer::unquantizeWemb<Type::int8>(items[i], ptr);
       } else {
+        resize(headers[i].dataLength);
         cpu::integer::prepareAndTransposeB<Type::int8>(items[i], ptr);
       }
-    } else if (matchType<intgemm16>(items[i].type)) {
-      if (items[i].name.find("Wemb") != std::string::npos) { // Since Wemb need to be dequantised, we have a special case for them
+    } else if(matchType<intgemm16>(items[i].type)) {
+      if(items[i].name.find("Wemb") != std::string::npos) {  // Since Wemb need to be dequantised,
+                                                             // we have a special case for them
         items[i].type = Type::float32;
-        items[i].bytes.resize(items[i].shape.elements()*sizeof(float)); // We should have an extra float at the back but that requires a different format, due to allocator work
+        resize(items[i].shape.elements()
+               * sizeof(float));  // We should have an extra float at the back but that requires a
+                                  // different format, due to allocator work
         cpu::integer::unquantizeWemb<Type::int16>(items[i], ptr);
       } else {
+        resize(headers[i].dataLength);
         cpu::integer::prepareAndTransposeB<Type::int16>(items[i], ptr);
       }
     } else {
-      std::copy(ptr, ptr + len, items[i].bytes.begin());
+      resize(headers[i].dataLength);
+      std::copy(ptr, ptr + headers[i].dataLength, items[i].bytes->begin());
     }
   }
+
+  LOG(info, "[memory] Model data loaded in: {}", totalBytesLoaded);
```

Comment:
This was originally a debug log. We should work on hooking up the logs to a proper pref so that we can get them.